### PR TITLE
Add commit message lint

### PR DESCRIPTION
**Pull request by narrowizard**
This PR added commit message lint([rules](https://www.conventionalcommits.org/en/v1.0.0/#summary)), including the following parts  
- [x] check commit message when commit.
- [x] check commit message on pull request checks (via github actions). 
- [x] add "git commit message section" on contributing guide.
- [x] add git commit tools for generating "conventional commit message".

> After this PR merged, commit message unsatisfied "conventional commit message" will be rejected. 

- [Why use conventional commits?](https://www.conventionalcommits.org/en/v1.0.0/#why-use-conventional-commits)

## Tool chains
For auto checking the commit message, we added some tools. now I'll explain the usage of each tool.

### github actions
The added github action [wagoid/commitlint-github-action@v4](https://github.com/wagoid/commitlint-github-action) is used for checking PR's commit message (each commit between target and source). 

### husky
[husky](https://github.com/typicode/husky) helps us using local git hooks in a easy way. we have set up a pre commit hook for checking commit message locally. That means the commit will be rejected if commit message is non-compliance.

### commitizen
[commitizen](https://github.com/commitizen/cz-cli) is a commit cli tool for git. you can use the cli(cz) to commit with a properly commit message. It's optional.

### commitlint
[commitlint](https://github.com/conventional-changelog/commitlint) helps us defining the commit message rules. we use the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard for now, maybe add more rules in the future.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,10 @@
     "enrich": "nodemon src/enrichment/main.js",
     "enrichment-worker": "./src/enrichment/worker.js",
     "lint": "./node_modules/eslint/bin/eslint.js . --max-warnings=0 --report-unused-disable-directives --fix && echo '✔  Your .js files look good.'",
-    "all": "concurrently \"nodemon src/collection/main.js\" \"nodemon src/collection/worker.js\" \"nodemon src/enrichment/main.js\" \"nodemon src/enrichment/worker.js\""
+    "all": "concurrently \"nodemon src/collection/main.js\" \"nodemon src/collection/worker.js\" \"nodemon src/enrichment/main.js\" \"nodemon src/enrichment/worker.js\"",
+    "prepare": "husky install",
+    "commit-msg": "commitlint -e $HUSKY_GIT_PARAMS",
```
Comment (UltimateBeaver) [inline package.json:17]: Could you please explain what `prepare` and `commit-msg` are for? Or when should we run these commands?

**Review by UltimateBeaver — COMMENTED** [review]
@narrowizard Thanks for this well-written PR. Some minor tweaks and we'll be able to merge this PR:

1. Could you please rebase this branch on top of the latest main branch so that we maintain a semi-linear commit history when we merge this PR? 

2. Could you please add a short paragraph in the PR description to talk about the tools we added in this PR? For example, what's the responsibility of each tool (husky, github actions, commitlint, commitizen) and what are their relationship.

File: package.json (hunk 1 above)
Comment (narrowizard) [reply package.json]: The `prepare` script in package.json is a npm hook, which will be auto triggered after `npm install`. `commit-msg` is deprecated, and removed now.

**Comment by narrowizard** [comment]
Ready to merge, comment if any more questions. @UltimateBeaver

**Comment by UltimateBeaver** [comment]
@narrowizard Thanks for the clarification. Merging now.

**Pull request merged**